labeling/codebase/localcodebase.go

```diff
@@ -14,23 +15,39 @@ type LocalCodebase struct {
 func (c LocalCodebase) FindFileMatching(
 	predicate func(string) bool,
 	glob ...string,
-) (string, error) {
-	for _, g := range glob {
-		filesFound, err := filepath.Glob(filepath.Join(c.BasePath, g))
-
-		if err != nil {
-			continue
-		}
-
-		for _, path := range filesFound {
-			relPath, err := filepath.Rel(c.BasePath, path)
-			if predicate(relPath) {
-				return relPath, err
+) (foundPath string, err error) {
+	basePath := c.BasePath
+	if basePath == "" {
+		basePath = "."
+	}
+	err = filepath.WalkDir(
```

Comment:
Should this error be checked and handled/returned?

Reply:
I was just lumping it together in the "not found" category, but you are right we should explicitly check it.